src/azure-cli/azure/cli/command_modules/containerapp/_params.py

```diff
@@ -0,0 +1,481 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long, too-many-statements, consider-using-f-string
+
+from knack.arguments import CLIArgumentType
+
+from azure.cli.core.commands.parameters import (resource_group_name_type,
+                                                get_location_type,
+                                                file_type,
+                                                get_three_state_flag, get_enum_type, tags_type)
+
+from ._validators import (validate_memory, validate_cpu, validate_managed_env_name_or_id, validate_registry_server,
+                          validate_registry_user, validate_registry_pass, validate_target_port, validate_ingress,
+                          validate_storage_name_or_id, validate_cors_max_age, validate_allow_insecure)
+from ._constants import UNAUTHENTICATED_CLIENT_ACTION, FORWARD_PROXY_CONVENTION, MAXIMUM_CONTAINER_APP_NAME_LENGTH, LOG_TYPE_CONSOLE, LOG_TYPE_SYSTEM
+
+
+def load_arguments(self, _):
+
+    name_type = CLIArgumentType(options_list=['--name', '-n'])
+
+    with self.argument_context('containerapp') as c:
+        # Base arguments
+        c.argument('name', name_type, metavar='NAME', id_part='name', help=f"The name of the Containerapp. A name must consist of lower case alphanumeric characters or '-', start with a letter, end with an alphanumeric character, cannot have '--', and must be less than {MAXIMUM_CONTAINER_APP_NAME_LENGTH} characters.")
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.ignore('disable_warnings')
+
+    with self.argument_context('containerapp') as c:
+        c.argument('tags', arg_type=tags_type)
+        c.argument('managed_env', validator=validate_managed_env_name_or_id, options_list=['--environment'], help="Name or resource ID of the container app's environment.")
+        c.argument('yaml', type=file_type, help='Path to a .yaml file with the configuration of a container app. All other parameters will be ignored. For an example, see  https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec#examples')
```

Comment:
why not combine them?

Reply:
It is OK to combine together to one argument_context.